plane/src/drone/runtime.rs

```diff
@@ -4,28 +4,35 @@ use crate::{
     types::BearerToken,
 };
 use anyhow::Error;
-use futures_util::Stream;
+use futures_util::{Future, Stream};
+use serde::{de::DeserializeOwned, Serialize};
 
-#[allow(async_fn_in_trait)]
```

Comment:
Short answer: we no longer need it because we no longer have any `async fn` methods in the trait.

Long answer: an `async fn` in Rust is just syntactic sugar for a function that returns a `Future` (similar to how in JS, it's just sugar for a function that returns a `Promise`). Normally when you define an `async fn` in Rust, the compiler determines whether the `Future` returned can safely be passed across threads (this boils down to: can the compiler ensure that everything that could be on the stack at the time of each `.await` call be safely passed across threads).

For `async fn` defined in a _public_ trait, the compiler doesn't have an implementation to infer that from; it needs to pick a function signature that applies to all `async fn` implementations (for a non-public trait, I guess it can check every implementation in the code and infer those bounds? I'm not sure if it actually does this, but the warning applies specifically to public traits).

In our case, we _do_ want the future to be passable between threads, so we need to desugar the trait methods from `async fn` to `fn ... -> Future<...> + Send`.